examples/05.HTTP_SERVER/xmake.lua

```diff
@@ -0,0 +1,62 @@
+-- Copyright CHERIoT Contributors.
+-- SPDX-License-Identifier: MIT
+
+-- Update this to point to the location of the CHERIoT SDK
+sdkdir = path.absolute("../../../cheriot-rtos/sdk")
+
+set_project("CHERIoT HTTP Server Example")
+
+includes(sdkdir)
+
+set_toolchains("cheriot-clang")
+
+includes(path.join(sdkdir, "lib"))
+includes("../../lib")
+
+option("board")
+  set_default("ibex-arty-a7-100")
+
+compartment("http_server_example")
+  set_default(false)
```

Comment:
I do not actually understand what the `set_default` does. I seem to understand that when passed `false` the compartment will not be compiled unless used (=depended on by something compiled)?

Reply:
Yes, but since this is a compartment that that is a direct dependency of the root target, it is not necessary. Use false when you are creating a library and want people to be able to include a load of libraries in xmake but not build them unless they’re used.